pkg/client/unversioned/clientcmd/loader.go

```diff
@@ -148,6 +164,36 @@ func NewDefaultClientConfigLoadingRules() *ClientConfigLoadingRules {
 	}
 }
 
+func loadFileOrDir(fileOrDir string) ([]string, error) {
+	stat, err := os.Stat(fileOrDir)
+	if err != nil {
+		return []string{}, err
+	}
+	if stat.IsDir() {
+		return loadConfigDir(fileOrDir)
+	}
+	return []string{fileOrDir}, nil
+}
+
+func loadConfigDir(dir string) ([]string, error) {
+	if len(dir) == 0 {
+		return []string{}, nil
+	}
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return []string{}, err
+	}
+	result := []string{}
+	for _, file := range files {
+		// Skip hidden files
```

Comment:
I'm with @justinsb here, I think this is unix specific (except for unix programs ported to windows which _do_ drop files starting with "."